File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/PresenterProvider.java
package ru.tutu.stations.ui.mvp;

import ru.tutu.stations.ui.mvp.presenter.MvpPresenter;

/**
 * @author Grigoriy Pryamov
 */
public interface PresenterProvider<P extends MvpPresenter> {
    P providePresenter();
}

File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/PresenterProvider.kt
package ru.tutu.stations.ui.mvp

import ru.tutu.stations.ui.mvp.presenter.MvpPresenter

/**
 * Фабрика презентеров
 *
 * @author Grigoriy Pryamov.
 */
interface PresenterProvider<P : MvpPresenter<MvpView>> {
    fun providePresenter(): P
}